### PR TITLE
feat: plugin/dependencies version control

### PR DESCRIPTION
**Pull request by Cdm2883**
# What does this PR do?

- [x] The plugins force the use of the https://semver.org/ specification version
- [x] Check if a plugin dependency version satisfies a requirement
- [x] Unit tests

### CODE REVIEW

**Comment by smartcmd** [comment]
喵喵喵

**Comment by Cdm2883** [comment]
> 喵喵喵

嗷嗷嗷

**Review by smartcmd — CHANGES_REQUESTED** [review]

File: server/src/main/java/org/allaymc/server/plugin/AllayPluginManager.java [hunk 1]
```diff
@@ -81,13 +83,24 @@ protected void onLoad(Map<String, PluginDescriptor> descriptors, Map<String, Plu
...
             for (var dependency : descriptor.getDependencies()) {
-                if (plugins.containsKey(dependency.name())) continue;
-
-                if (!dependency.optional()) {
-                    log.error(I18n.get().tr(TrKeys.A_PLUGIN_DEPENDENCY_MISSING, descriptor.getName(), dependency.name()));
-                    iterator.remove();
+                var dependencyContainer = plugins.get(dependency.name());
+
+                if (dependencyContainer == null) {
+                    if (!dependency.optional()) {
```
Comment (smartcmd) [inline server/src/main/java/org/allaymc/server/plugin/AllayPluginManager.java:89]: 感觉这个嵌套if可以合并成一个if(

File: server/src/test/java/org/allaymc/server/plugin/AllayPluginManagerTest.java [hunk 2]
```diff
@@ -17,6 +20,11 @@
  * @author daoge_cmd
  */
 public class AllayPluginManagerTest extends AllayPluginManager {
+    @BeforeAll
+    static void initI18n() {
```
[the remark this answers is not in the document]
Comment (Cdm2883) [reply server/src/test/java/org/allaymc/server/plugin/AllayPluginManagerTest.java:24]: But AllayTestExtension will load unnecessary things like the Allay server itself.

**Review by smartcmd — CHANGES_REQUESTED** [review]

File: server/src/main/java/org/allaymc/server/plugin/AllayPluginManager.java (hunk 1 above)
Comment (smartcmd) [inline server/src/main/java/org/allaymc/server/plugin/AllayPluginManager.java:89]: Use guard-statement

**Review by smartcmd — CHANGES_REQUESTED** [review]

File: server/src/main/java/org/allaymc/server/plugin/AllayPluginManager.java (hunk 1 above)
Comment (Cdm2883) [reply server/src/main/java/org/allaymc/server/plugin/AllayPluginManager.java:89]: Sorry for the mindless copy and paste, I didn't see it qwq

**Review by smartcmd — APPROVED** [review]

**Pull request merged**